File: 1. Programming Languages/C++/01. OOPs/06. Encapsulation/01_Basics.cpp
/*
Encapsulation: 
-> Wrapping up of data and information in a single unit, while controlling access to them 
*/

#include <iostream>
using namespace std;

class Customer
{
    string name;
    int balance;

public: 
    int age;

    Customer(string a, int b)
    {
        name = a;
        balance = b;
        age = 21;
    }

    void display()
    {
        cout << "Name: " << name << endl;
        cout << "Balance: " << balance << endl;
        cout << "Age: " << age << endl;
    }
};

int main()
{
    Customer C1("Karan", 10000);
    C1.age = -4;

    C1.display(); 
    //! Note: Age can't be negative but here it is that's why we should keep our crucial data in private section only
}

File: 6. MERN/4. JavaScript/05. Arrays/01_Basics.js
let arr = [10, 20.4, true, false, -34, -54.53, null, 0/0, "Karan", NaN, {"age": 21}];

console.log(arr);         // [10, 20.4, true, false, -34, -54.53, null, undefined, "Karan", NaN, {"age": 21}]
console.log(arr.length);  // 11
console.log(arr[0]);      // 10
console.log(arr[1]);      // 20.4
console.log(arr[2]);      // true
console.log(arr[3]);      // false
console.log(arr[11]);     // undefined
console.log(arr[-2]);     // undefined

arr[0] = 100;

console.log(arr[0]);      // 100